Minor README fixes (#2)
Sporarum/GeodeLoopless: README.md

Geode Loopless (GeodeL) is a Domain Specific Language that aims at facilitating the writing and analysis of Primitive Recursive Functions and Sets.

This DSL is embedded in Scala, and aims to use its strong type system to enforce the arity and type of the `PrimRec` functions and sets, thus ensuring any function or set is really primitive recursive.

> Primitive recursive functions are all the functions that can be created by combining the successor function on naturals and constants with projection, recursion, and composition.

Reviewer: ```suggestion
Primitive recursive functions are all the functions on naturals that can be created by combining the successor function and constants with projection, recursion, and composition.
```
Or maybe
```suggestion
Primitive recursive functions are all the functions on tuples of naturals that can be created by combining the successor function and constants with projection, recursion, and composition.
```

Reviewer: Also what do you think of this ?
```suggestion
Primitive recursive functions are all the functions that can be created by combining constants, projections, and the successor function through recursion and composition.
```

Author: ```suggestion
Primitive recursive functions are all the functions on naturals that can be created by combining constants, projections, and the successor function through recursion and composition.
```

I like the combination of changes 1 and 3 here. It's the clearest of them.

Reviewer: Yes I agree, these were intended to be combined !

> In Scala, this translates to the following ([the actual implementation](https://github.com/Sporarum/GeodeLoopless/blob/main/src/main/scala/PrimRecFun.scala) also includes a "stack trace" method `debug`):

Reviewer: ```suggestion
In Scala, this translates to the following ([the actual implementation](src/main/scala/PrimRecFun.scala) also includes a "stack trace" method `debug`):
```
We can do it here as well, no ?

Author: This was the one I suggested (line 12). I did not do it myself because I thought I'd get your opinion on it. It's a somewhat subjective choice.

Reviewer: Yeah I see, I think it makes sense for this one, but not for the things in `DerivedFunctions`
